### PR TITLE
Feature/fix rate limiter bug

### PR DESCRIPTION
**Pull request by ronsherf**
@barnash @shaharke @yardenmintz

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shaharke) [inline .idea/vcs.xml]: Don't check in this file..

File: test/index.spec.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 var chai = require('chai');
 var expect = chai.expect;
-
+var async = require('async');
```
Comment (shaharke) [inline test/index.spec.js]: Why are we using `async`? There's nothing here we can't implement with promises..

File: test/index.spec.js [hunk 3]
```diff
@@ -20,9 +19,9 @@ describe('Rate limit wrapper', function() {
 
   beforeEach(function(done) {
     client.keys('limit:*', function(err, rows) {
-      async.each(rows, function(row, callbackDelete) {
-        client.del(row, callbackDelete);
-      }, done);
+      Q.all(rows.map(function(row){
```
Comment (shaharke) [inline test/index.spec.js]: With `mocha` you can get rid of the `done` callback function

File: .idea/vcs.xml (hunk 1 above)
Comment (ronsherf) [inline .idea/vcs.xml]: Fixed it together.

**Comment by ronsherf** [comment]
@shaharke 
1. Fixed the merge issue.
2. Since re-writing the test without done callback function require test flow changes, decided with you to revert to the last working commit.

**Pull request merged**